gtsfm/densify/mvs_base.py

```diff
@@ -0,0 +1,62 @@
+"""Base class for the MVS stage of the back-end.
+
+Authors: John Lambert, Ren Liu
+"""
+import abc
+from typing import Dict, Tuple
+
+import dask
+import numpy as np
+from dask.delayed import Delayed
+from gtsam import PinholeCameraCal3Bundler
+
+from gtsfm.common.image import Image
+
+
+class MvsBase(metaclass=abc.ABCMeta):
+    """Base class for all multi-view stereo hyperparameters."""
+
+    def __init__(self) -> None:
+        """Initialize the MVS module """
+        pass
+
+    @abc.abstractmethod
+    def densify(
+        self,
+        images: Dict[int,Image],
+        cameras: Dict[int, PinholeCameraCal3Bundler],
+        min_distance: float,
+        max_distance: float,
+    ) -> Tuple[np.ndarray, Dict[int, np.ndarray]]:
```

Comment:
so this won't take sparse point cloud as input?

Reply:
There are a few ways to go about this. One is compute a single min and max depth for the whole scene. But for a big scene, and a long trajectory, I guess that doesnt make sense, since that can vary dramatically per camera. So maybe we should be computing the min and max depth per camera.? @dellaert  @codyly